ompi/request/req_wait.c

```diff
@@ -384,12 +401,16 @@ int ompi_request_default_wait_some(size_t count,
             /* If the request is completed go ahead and mark it as such */
             assert( REQUEST_COMPLETE(request) );
             num_requests_done++;
+            indices[i] = 1;
```

Comment:
Instead of the memset and the assignment to indices[i], you can use the indices array to store directly the outcome of the atomic operation, so that the if on line 384 transforms in ```if( !(indices[i] = (int)OPAL_ATOMIC_CMPSET_PTR(&request->req_complete, REQUEST_PENDING, &sync)) ) {```